js-pkg/driftdb-react/src/index.tsx

```diff
@@ -246,3 +246,22 @@ export function DriftDBProvider(props: DriftDBProviderProps) {
 
     return <DatabaseContext.Provider value={dbRef.current}>{props.children}</DatabaseContext.Provider>;
 }
+
+export function useSubscription<T>(key: string, callback: (data: any) => void): (v: T) => void {
+    const db = useDatabase()
+    const subscription = useRef(false)
+
+    if (subscription.current === false) {
+        subscription.current = true
+        db.subscribe(key, callback)
+    }
```

Comment:
I'm wondering if this should be a `useEffect` instead of using `useRef`. The subscription ref is just used as a flag to perform a side effect in a render fn (which shouldn't have side effects). 

Also, if the callback changes, the `db` will still be subscribed with the old callback. So if someone does something like this, they will get the callback fired with stale props values:

```ts
function MyComponent (props) {
  useSubscription('my-key', () => {
    // use some props here
  })
}
```

You could also have a situation where the callback doesn't fire at all if the key changes:

```ts
function MyComponent (props) {
  useSubscription(props.key, myCallback)
}
```

Also, when the component goes away, should this be subscription be dropped?

I think `useEffect` might address all these issues?:

```ts
useEffect(() => {
  db.subscribe(key, callback)
  return () => {
    db.unsubscribe(key, callback)
  }
}, [key, callback])
```